unified-runtime/source/loader/layers/sanitizer/msan/msan_interceptor.hpp

```diff
@@ -148,8 +148,71 @@ struct ContextInfo {
   ~ContextInfo();
 };
 
+struct MsanRuntimeDataWrapper {
+  MsanRuntimeData Host{};
+
+  MsanRuntimeData *DevicePtr = nullptr;
+
+  ur_context_handle_t Context{};
+
+  ur_device_handle_t Device{};
+
+  MsanRuntimeDataWrapper(ur_context_handle_t Context, ur_device_handle_t Device)
+      : Context(Context), Device(Device) {}
+
+  ~MsanRuntimeDataWrapper();
+
+  MsanRuntimeData *getDevicePtr() {
+    if (DevicePtr == nullptr) {
+      ur_result_t Result = getContext()->urDdiTable.USM.pfnDeviceAlloc(
+          Context, Device, nullptr, nullptr, sizeof(MsanRuntimeData),
+          (void **)&DevicePtr);
+      if (Result != UR_RESULT_SUCCESS) {
+        getContext()->logger.error(
+            "Failed to alloc device usm for asan runtime data: {}", Result);
```

Comment:
```suggestion
            "Failed to alloc device usm for msan runtime data: {}", Result);
```

TODO: we can make this helper class a unified one for all Sanitizers.

Reply:
Done